chainstate/src/lib.rs

```diff
@@ -14,6 +14,7 @@
 // limitations under the License.
 
 pub mod chainstate_interface;
+pub mod chainstate_interface_impl_delegation;
```

Comment:
why not chainstate_interface_delegation?

Reply:
Because we're delegating the implementation, not the interface.